src/api/spec/factories/notification.rb

```diff
@@ -23,104 +32,108 @@
+      end
     end
 
-    trait :comment_for_request do
-      event_type { 'Event::CommentForRequest' }
-      notifiable factory: [:comment_request]
-    end
+    factory :notification_for_comment, class: 'NotificationComment' do
+      trait :comment_for_project do
+        event_type { 'Event::CommentForProject' }
+        notifiable factory: [:comment_project]
+      end
 
-    trait :relationship_create_for_project do
-      event_type { 'Event::RelationshipCreate' }
-      notifiable factory: [:project]
-    end
+      trait :comment_for_package do
+        event_type { 'Event::CommentForPackage' }
+        notifiable factory: [:comment_package]
+      end
 
-    trait :relationship_delete_for_project do
-      event_type { 'Event::RelationshipDelete' }
-      notifiable factory: [:project]
-    end
-    trait :relationship_create_for_package do
-      event_type { 'Event::RelationshipCreate' }
-      notifiable factory: [:package]
+      trait :comment_for_request do
+        event_type { 'Event::CommentForRequest' }
+        notifiable factory: [:comment_request]
+      end
     end
 
-    trait :relationship_delete_for_package do
-      event_type { 'Event::RelationshipDelete' }
-      notifiable factory: [:package]
-    end
+    factory :notification_for_project, class: 'NotificationProject' do
+      trait :relationship_create_for_project do
+        event_type { 'Event::RelationshipCreate' }
+        notifiable factory: [:project]
+      end
 
-    trait :build_failure do
-      event_type { 'Event::BuildFail' }
-      notifiable factory: [:package]
+      trait :relationship_delete_for_project do
+        event_type { 'Event::RelationshipDelete' }
+        notifiable factory: [:project]
+      end
     end
 
-    trait :create_report do
-      event_type { 'Event::CreateReport' }
-      notifiable factory: [:report]
+    factory :notification_for_package, class: 'NotificationPackage' do
+      trait :relationship_create_for_package do
+        event_type { 'Event::RelationshipCreate' }
+        notifiable factory: [:package]
+      end
 
-      transient do
-        reason { nil }
+      trait :relationship_delete_for_package do
+        event_type { 'Event::RelationshipDelete' }
+        notifiable factory: [:package]
       end
 
-      after(:build) do |notification, evaluator|
-        notification.event_payload['reportable_type'] ||= notification.notifiable.reportable.class.to_s
-        notification.event_payload['reason'] ||= evaluator.reason
+      trait :build_failure do
+        event_type { 'Event::BuildFail' }
+        notifiable factory: [:package]
       end
     end
 
-    trait :cleared_decision do
-      event_type { 'Event::ClearedDecision' }
-      notifiable { association(:decision_cleared) }
+    factory :notification_for_report, class: 'NotificationReport' do
+      trait :create_report do
+        event_type { 'Event::CreateReport' }
+        type { 'NotificationReport' }
```

Comment:
```suggestion
    factory :notification_for_report, class: 'NotificationReport' do
      trait :create_report do
        event_type { 'Event::CreateReport' }
```
The `type { 'NotificationReport' }` line looks redundant to me. In other sections of this same file, defining a `factory` with a `class` attribute is enough, and does not include a `type` line.